sympy/polys/tests/test_polyroots.py

```diff
@@ -73,6 +73,13 @@ def test_roots_quadratic():
         roots = roots_quadratic(f)
         assert roots == _nsort(roots)
 
+
+def test_issue_7724():
+    eq = Poly(x**4*I + x**2 + I, x)
+    roots_d = roots(eq)
+    assert len(roots_d) == 4
```

Comment:
I think it would be better to assert what the roots are then just how many there are

Reply:
Ok... I'll commit it asap